src/dialog.jsx

```diff
@@ -145,13 +149,19 @@ let Dialog = React.createClass({
   componentWillReceiveProps (nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
     this.setState({muiTheme: newMuiTheme});
+
+    if( nextProps.isOpen !== this.props.isOpen ) {
+      if( nextProps.isOpen && !this.state.open )
+        this._show();
+      else if( !nextProps.isOpen && this.state.open )
+        this._dismiss();
+    }
   },
 
   componentDidMount() {
     this._positionDialog();
     if (this.props.openImmediately) {
       this.refs.dialogOverlay.preventScrolling();
```

Comment:
Didn't want to move this out since it needs `nextProps`, feels wrong to start passing that along.
To keep this part compact I preserved the `_show` and `_dismiss` methods.